core.js

```diff
@@ -0,0 +1,685 @@
+      if (!disassociated){
+        //TODO: log warning?
+      }
+      //now go through each connection associated with this publisher
+      for(connectionI = publisher.connectedTo.length - 1;
+          connectionI >= 0;
+          connectionI--){
+        connection = publisher.connectedTo[connectionI];
+        disassociated = false;
+        //for each route associated with this connection
+        for(connectRouteI = connection.routes.length - 1;
+            connectRouteI >= 0 && !disassociated;
+            connectRouteI--){
+          //if the connection is associated with this route
+          if (connection.routes[connectRouteI] === route){
+            //remove the association
+            connection.routes.splice(connectRouteI, 1);
+            //there should only be one association
+            disassociated = true;
+            //if there are no other associations
+            if (connection.routes.length === 0){
+              //then we should break the connection
+              Manager.breakConnection(publisher, connection);
+            }
+          }
+        }
+      }
+    }
+    this.routes.splice(routeI, 1);
+
+    // tell all the admins about the removed route and connections
+    for(var adminI = this.admins.length - 1; adminI >= 0; adminI--){
+      this.admins[adminI].sendRemove(update.clients,
+                                     update.routes,
+                                     update.connections);
+    }
+  }
+  return (routeI >= 0);
+};
+
+/**
+ * Gets an array detailing the routes currently registered with this server.
+ * @returns {Array} An array of route info: 
+ *   [{type:string, style:string, uuid:string, 
+ *     from:{name:string, metadata:object, endpoint:string}, 
+ *     to:{name:string, metadata:object, endpoint:string}},...]
+ */
+Manager.prototype.getRoutes = function(){
+  var routes = [];
+  for(var i = 0; i < this.routes.length; i++){
+    var curr = this.routes[i];
+    routes.push(curr.toMap());
+  }
+  return routes;
+};
+
+/**
+ * Gets an array of specific publisher/subscriber 
+ *   connections that currently exist.
+ * @returns {Array} an array of pub/sub connection info:
+ *   [{type:string, from:{uuid:string, endpoint:string}, 
+ *     to:{uuid:string, endpoint:string}, routes:[string,...]}]
+ */
+Manager.prototype.getConnections = function(){
+  var connections = [];
+  //the connections are stored as references from each publisher to subscriber
+  
+  //for each client
+  for(var pubClientI = this.clients.length - 1;
+      pubClientI >= 0;
+      pubClientI--){
+    var pubClient = this.clients[pubClientI];
+    //get all the connections coming from that client
+    var clientConnections = pubClient.getOutgoingConnections();
+    //and add them to the list
+    connections = connections.concat(clientConnections);
+  }
+
+  return connections;
+};
+
+Manager.prototype.published = function(pubClient, pubName, type, message){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function 'anonymous' has a complexity of 7.](https://www.codacy.com/app/quinkennedy/spacebrew/file/2279533613/issues/source?bid=3184377&fileBranchId=3187330#l649)